### PR TITLE
Experiment: installer-help redesign. (#8602)

### PR DESCRIPTION
**Pull request by amychurchwell**
## Description
Testing conversion rate of a redesigned firefox/installer-help page.

## Issue / Bugzilla link
#8602

## Testing
**demo:** https://www-demo5.allizom.org/en-US/firefox/installer-help/?v=b

- [ ] **[Test plan](https://docs.google.com/document/d/1nI6GlQjZ3mH7_J_Dsdgpd2Ia7hke0lDxgwT7BrvvJwU/edit?usp=sharing)**: 54% of users entered into experiment total. Half of those receive control and half receive variation (27% each)
- [ ] If no channel passed users receives all download buttons and updated copy.
- [ ] If channel passed user receives appropriate download button.

### CODE REVIEW

**Comment by amychurchwell** [comment]
Question for reviewer: should [custom stub attribution](https://bedrock.readthedocs.io/en/latest/stub-attribution.html#measuring-campaigns-and-experiments) also be a part of this experiment?

File: bedrock/firefox/templates/firefox/installer-help-b.html [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  {% set channel_logo = 'protocol/img/logos/firefox/browser/nightly/logo-word-hor-sm.png' %}
+{% else %}
+  {% set channel_logo = 'protocol/img/logos/firefox/browser/logo-word-hor-sm.png' %}
+{% endif %}
+
+{% block content %}
+<main class="mzp-l-content">
+  <section class="mzp-c-hero">
+    <div class="mzp-c-hero-body">
+      <h1 class="mzp-c-hero-title">{{ _('Hmm... your download was interrupted') }}</h1>
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/installer-help-b.html]: Nit: should this be a an ellipsis character (`…`) instead of three periods? (`...`)

File: bedrock/firefox/templates/firefox/installer-help-b.html [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            <span>{{ _('Choose a download below.') }}</span>
+          {% endif %}
+        </p>
+      </div>
+    </div>
+  </section>
+
+  {% if installer_channel %}
+    <section class="installer-channel-card mzp-c-card mzp-c-card-extra-small">
+      {{ high_res_img(channel_logo, {'alt': 'Firefox', 'width': '347', 'height': '64'}) }}
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/installer-help-b.html]: Nit: should the alt text for each logo more accurately reflect the words in each image?

File: bedrock/firefox/templates/firefox/installer-help-b.html [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+          {{ download_firefox('nightly', platform='desktop', force_direct=True, force_full_installer=True, locale=installer_lang, button_color='mzp-t-secondary mzp-t-small', alt_copy=_('Download Now')) }}
+        </div>
+      </section>
+    </div>
+  {% endif %}
+
+  <section class="install-help">
+    <div>
+      <h2>{{ _('Need help installing?') }}</h2>
+      <p>{{ _('If you still need help installing, <a href="https://support.mozilla.org/kb/install-firefox-windows">read the tutorial in our Support section.</a>') }}</p>
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/installer-help-b.html]: Nit: link `href` should be passed to the translatable string as a variable

File: media/js/firefox/installer-help-experiment.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    var href = window.location.href;
+    if (href.indexOf('v=') !== -1) {
+        if (href.indexOf('v=a') !== -1) {
+            window.dataLayer.push({
+                'data-ex-variant': 'installer-help-control',
+                'data-ex-name': 'CRO-Installer-Help-Experiment'
+            });
+        } else if (href.indexOf('v=b') !== -1) {
+            window.dataLayer.push({
+                'data-ex-variant': 'installer-help-v1',
```
Comment (alexgibson) [inline media/js/firefox/installer-help-experiment.js]: Nit: should this be `vb` to reflect `v=b` in the URL?

File: media/static-bundles.json [hunk 5]
```diff
@@ -1406,6 +1412,12 @@
         "js/base/error-pages.js"
       ],
       "name": "page_not_found"
+    },
+    {
+      "files": [
+        "js/firefox/installer-help-experiment.js"
```
Comment (alexgibson) [inline media/static-bundles.json]: The page is throwing a JS error because Traffic Cop is undefined:

```
Uncaught TypeError: Mozilla.TrafficCop is not a constructor
```

File: media/js/firefox/installer-help-experiment.js (hunk 4 above)
Comment (amychurchwell) [reply media/js/firefox/installer-help-experiment.js]: I collaborated with Saptarshi and he requested this value. 👍

File: media/static-bundles.json (hunk 5 above)
Comment (amychurchwell) [reply media/static-bundles.json]: whoops! 😅

File: bedrock/firefox/templates/firefox/installer-help.html [hunk 6]
```diff
@@ -67,3 +67,7 @@ <h3 class="installer-help-emphasis-title">{{ _('Need help installing?') }}</h3>
 {% endblock %}
 
 {% block js %}{% endblock  %}
+
+{% block experiments %}
+  {{ js_bundle('installer_help_experiment') }}
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/installer-help.html:72]: Why did you remove the switch here? I'd suggest we keep it, so we can turn the experiment on/off independent of a code review and production push?

File: media/static-bundles.json [hunk 7]
```diff
@@ -1406,6 +1412,13 @@
         "js/base/error-pages.js"
       ],
       "name": "page_not_found"
+    },
+    {
+      "files": [
+        "js/firefox/installer-help-experiment.js",
+        "js/libs/mozilla-traffic-cop.js"
```
Comment (alexgibson) [inline media/static-bundles.json:1419]: Seems this is still throwing an error here, because you added traffic cop after the script that tries to use it.

File: bedrock/firefox/templates/firefox/installer-help.html (hunk 6 above)
Comment (amychurchwell) [reply bedrock/firefox/templates/firefox/installer-help.html:72]: I had removed it when I had pushed this to a demo branch for design review & I forgot to put it back in. Adding back now.

Comment (alexgibson) [reply bedrock/firefox/templates/firefox/installer-help.html]: Ah, demos are treated like `Dev=True`, so switches are on by default when you push there. It should "just work" ™️ without needing to remove.
